### PR TITLE
Bundle with Rollup

### PR DESCRIPTION
**Pull request by shvaikalesh**
# Tooling changes

* Replace [ym](https://github.com/ymaps/modules) with ES6 modules and bundle with Rollup
* Use more ES6 in more places
* Use `KeyboardEvent#key`, add [shim](https://github.com/shvaikalesh/shim-keyboard-event-key) for IE and Edge
* ESLint: replace standard with airbnb (latter has more rules and enforces more consistent code style)
* ESLint: use `yml` for configs instead of `json`
* Use better `.editorconfig` linting solution
* `EventGroup` now passes `priority` to `EventEmitter`
* Replace Gulp with npm scripts
* Run unit tests with jsdom
* Run functional tests in parallel
* Add test coverage for plugins

# User-facing changes

* Fix `data-timing` interop with inner navigation
* Add new duration format for `data-timing`
* Fix triggering inner navigation on `prev` when in list mode
* `#` does not get appended in `list` mode in Firefox
* `next` plugin uses `slide_active_classname` option for active items
* Add `CHANGELOG.md`

# Tests

* Safari fails because v10 does not support `KeyboardEvent#key` (v10.1 does, but it is not available in Sauce Labs) and we do not support Safari 10
* Edge fails on 1-2 `next` plugin tests for some weird reason, it can't locate elements that are definitely there. Sometimes it passes if one adds huge pauses between assertions

# Important

* Plugins are currently moved to this repo. Before we implement new plugin loading solution, custom plugins (that are not built-in) do not work.

Closes https://github.com/shower/shower/issues/253
Closes https://github.com/shower/shower/issues/257

### CODE REVIEW

**Comment by pepelsbey** [comment]
While we’re at it: I remember there was one misconception in JS implementation. In HTML modes are marked with `full` and `list`:

```html
<body class="shower full">
<body class="shower list">
```

Which makes them keywords for modes. But in JS there’s `slideMode` idea, that I’d rather call `fullMode`. This includes: `enterSlideMode`, `exitSlideMode` and maybe something else.

Do you agree?

**Comment by shvaikalesh** [comment]
Totally, was going to propose myself. Both modes has slides, thus `isSlideMode` is not as descriptive as it could be.

However, I think it belongs in separate PR, right next to `Container`/`Player` refactoring, getters/setters adoption etc. This PR is already quite big and covers more than it's title says.

File: lib/plugins/next.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    _getElements() {
+        const { layout } = this._shower.player.getCurrentSlide();
+        const slideEl = layout.getElement();
+
+        return Array.from(slideEl.querySelectorAll(this._elementsSel));
+    }
+
+    _getInnerComplete() {
+        return this._elements
+            .filter(el => el.classList.contains('active'))
```
Comment (shvaikalesh) [inline lib/plugins/next.js]: Should we get class for `active` state from options as we do for slides?

Comment (pepelsbey) [inline lib/plugins/next.js]: Configurable plugins is a good thing, I guess.

**Comment by pepelsbey** [comment]
Maybe it’s time to add `package-lock.json` to version control?

```
$ git status
On branch es-modules
Your branch is up-to-date with 'origin/es-modules'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	package-lock.json

nothing added to commit but untracked files present (use "git add" to track)
```

File: lib/emitter/event-group.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ *      this._shower = shower;
+ *
+ *      this._message = "Hello";
+ *      this._showerListeners = null;
+ * }
+ *
+ * MyClass.prototype.setupListeners = function () {
+ *      this._showerListeners = this._shower.events.group()
+ *          .on("next", function () { console.log(this._message); }, this)
+ *          .on("prev", function () { console.log(this._message); }, this);
```
Comment (shvaikalesh) [inline lib/emitter/event-group.js]: Me too, but we don't lint comments and this whole class will be dropped in the next PR.

File: lib/shower/shower.js [hunk 3]
```diff
@@ -200,15 +200,15 @@ class Shower {
...
-        const lr = document.createElement('section');
-        lr.className = 'region';
-        lr.setAttribute('role', 'region');
-        lr.setAttribute('aria-live', 'assertive');
-        lr.setAttribute('aria-relevant', 'additions');
-        lr.setAttribute('aria-label', 'Slide Content: Auto-updating');
-
-        this._liveRegion = lr;
-        document.body.appendChild(lr);
+        const region = document.createElement('section');
```
Comment (shvaikalesh) [inline lib/shower/shower.js]: Long variable names look extremely noisy in some cases. But whatever, as you wish.

Comment (pepelsbey) [inline lib/shower/shower.js]: You can always use `with` statement :D

**Comment by pepelsbey** [comment]
LGTM

**Pull request merged**